packages/shorebird_cli/lib/src/shorebird_build_mixin.dart

```diff
@@ -182,4 +184,54 @@ mixin ShorebirdBuildMixin on ShorebirdCommand {
       );
     }
   }
+
+  Future<String> createDiff({
+    required String releaseArtifactPath,
+    required String patchArtifactPath,
+  }) async {
+    final tempDir = await Directory.systemTemp.createTemp();
+    final diffPath = p.join(tempDir.path, 'diff.patch');
+    final diffExecutable = p.join(
+      cache.getArtifactDirectory('patch').path,
+      'patch',
+    );
+    final diffArguments = [
+      releaseArtifactPath,
+      patchArtifactPath,
+      diffPath,
+    ];
+
+    final result = await process.run(
+      diffExecutable,
+      diffArguments,
+      runInShell: true,
+    );
+
+    if (result.exitCode != 0) {
+      throw Exception('Failed to create diff: ${result.stderr}');
+    }
+
+    return diffPath;
+  }
+
+  Future<File> buildAotSnapshot({required String appDillPath}) async {
```

Comment:
Specifically this is building an elf aot snapthot (there are other kinds).

Reply:
Clarified